### DISCUSSION TITLE
Sub-menu in Toolbox

### DISCUSSION THREAD

**Discussion by piltsen** [codex-team/editor.js#1811] [Ideas]
We started using Editorjs and we are really enjoying the simplicity of it. 

But we realised that we are going to add a lot of plugins, and when we are adding a new item (pressing Tab on a new line), the list of icons is growing almost out of the screen. 

So I was wondering if perhaps there was a way to add a submenu, similar to the "convert to", in order to organise all the icons in the list. 

What do you reckon?

**Comment by neSpecc** [comment]
You're right. Horizontal Toolbox is not scalable and understandable.

We are working on the vertical Toolbox, that will look like this:

<img width="855" alt="image" src="https://user-images.githubusercontent.com/3684889/147080205-b2a67d98-8eb7-4e7e-8a79-7159841e1808.png">

**Reply by m-collins** [reply]
This looks great @neSpecc Do you have any idea of when this would be released

**Reply by anuragvohraec** [reply]
> You're right. Horizontal Toolbox is not scalable and understandable.
> 
> We are working on the vertical Toolbox, that will look like this:

Please do consider mobile devices in your designs.
Verticle toolbox can be problematic on the last line, so do consider it too.

**Comment by anuragvohraec** [comment]
As for now only adding a single css property will solve the issue, please can some one add this line of code:
![Fix](https://user-images.githubusercontent.com/53807480/159679309-184f38e4-a1e1-479b-a420-382c0eb7f54b.png)

**Comment by abuldevx** [comment]
Thank you @neSpecc for adding the `vertical Toolbox` feature. But If I want to use `horizontal Toolbox` there is no option available for me. So please add a config option so I can use the `horizontal` or `vertical` option.

**Comment by neSpecc** [comment]
Seems outdated
